src/outputs/restart_hdf5.cpp

```diff
@@ -134,6 +134,16 @@ RestartReaderHDF5::MeshInfo RestartReaderHDF5::GetMeshInfo() const {
   mesh_info.level_gid_lid_cnghost_gflag =
       ReadDataset<int>("/Blocks/loc.level-gid-lid-cnghost-gflag");
 
+  try {
+    mesh_info.derefinement_count = ReadDataset<int>("/Blocks/derefinement_count");
+  } catch (const std::runtime_error &e) {
```

Comment:
knowing HDF5 I bet there's voluminous output here... I'm fine with this solution, but if we wanted to reduce the amount that HDF5 complains, you could instead check for the existence of `"/Blocks/derefinement_count"` with, e.g., `H5Lexists` (https://docs.hdfgroup.org/hdf5/v1_12/group___h5_l.html#title11). Despite the function being  documented as looking at links, it works for any "name" in an HDF5 file: https://forum.hdfgroup.org/t/check-if-dataset-exists/1454/5

Reply:
Honestly, I am a little confused by the output. The beginning of `ReadDataset` does 
```c++
    // make sure dataset exists
    auto status = PARTHENON_HDF5_CHECK(H5Oexists_by_name(fh_, name.c_str(), H5P_DEFAULT));
    PARTHENON_REQUIRE_THROWS(
        status > 0, "Dataset '" + name + "' does not exist in HDF5 file " + filename_);
```
and that throw is what I am catching here I think.

Reply:
Ah. That makes sense. In which case yeah I wouldn't expect HDF5 to complain at all. Bizarre.

Reply:
yeah in that case disregard my suggestion.

Reply:
I am going to leave it as is for now.

Reply:
Given that the (quite extensive) error message is printed by every rank, I now went for the `H5Lexists` option as @Yurlungur suggested (and we already use for the `SparseInfo` field).